Add read/write support for map stats

sdruzkin opened the pull request:
Update DWRF metadata reader and writer to read and write map statistics from the file.

This change will make the reader read map stats from the file. Flat map column writer map stats are not wired yet, so the writer isn't going to change the behavior.

Wiring map stats and the new flag in the flat map column writer is going to be the last diff in this series of changes to enable map stats.

Test plan:
- new unit tests
- Ran Uncle Vader for flat map enabled tables, results can be found under presto_map_stats_rw tag


```
== NO RELEASE NOTE ==
```

sdruzkin commented on presto-orc/src/test/java/com/facebook/presto/orc/metadata/TestDwrfMetadataReader.java:
```diff
@@ -333,4 +338,35 @@ private static StringStatistics createExpectedStringStatistics(HiveWriterVersion
+        mapStatsStringKeyBuilder.addMapStatistics(DwrfProto.KeyInfo.newBuilder().setBytesKey(ByteString.copyFromUtf8("k2")).build(), integerColumnStatistics2);
+        ColumnStatistics mapColumnStatisticsStringKey = mapStatsStringKeyBuilder.buildColumnStatistics();
+
+        return new Object[][] {
```
This can be extend for other types of stats.

arunthirupathi commented:
Do we have a use case to read this ? If not, we can avoid adding it for now or put it behind a flag. Some of the map statistics are huge and it wastes lot of memory.

The PR looks good.

arunthirupathi commented:
> Makes sense to disable reading map stats by default. To do this we would need to add a new field to OrcReaderOptions and then propagate OrcReaderOptions from the OrcReader to the DwrfMetadataReader. This will require changing the API of MetadataReader and StripeMetadataSource, which is 5x work. I can do it tomorrow.

Let us keep this PR open and not merge it. We also don't need to work on the configuration property for now.

sdruzkin commented:
1. Added readMapStats field to the OrcReaderOptions.
2. Plugged OrcReaderOptions as a constructor argument of the DwrfMetadataReader. This allows to avoid cascading API changes.
3. Because of N2 I switched some methods in DwrfMetadataReader from static to instance.

sdruzkin commented:
An unrelated test failed, re-triggering the build.
